src/Databases/DatabaseOnDisk.cpp

```diff
@@ -745,8 +765,9 @@ ASTPtr DatabaseOnDisk::parseQueryFromMetadata(
             metadata_file_path);
     }
 
-    ReadBufferFromFile in(metadata_file_fd, metadata_file_path, METADATA_FILE_BUFFER_SIZE);
-    readStringUntilEOF(query, in);
+    auto read_buf = shared_disk->readFile(metadata_file_path, ReadSettings{});
```

Comment:
It is not good to use default read settings, use `getReadSettings()` from `IO/ReadSettings.h`.
Also please set `METADATA_FILE_BUFFER_SIZE` to `read_settings` as `read_settings.adjustBufferSize(METADATA_FILE_BUFFER_SIZE)`

Reply:
I will update it.